Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -1472,15 +1472,9 @@ bool GraphicsContextCG::isCALayerContext() const
     return m_data && m_data->m_contextFlags.contains(GraphicsContextCGFlag::IsLayerCGContext);
 }
 
-void GraphicsContextCG::setIsAcceleratedContext(bool isAccelerated)
-{
-    // Should be called for CA Context.
-    m_data->m_contextFlags.set(GraphicsContextCGFlag::IsAcceleratedCGContext, isAccelerated);
-}
-
 RenderingMode GraphicsContextCG::renderingMode() const
 {
-    return m_data->m_contextFlags.contains(GraphicsContextCGFlag::IsAcceleratedCGContext) ? RenderingMode::Accelerated : RenderingMode::Unaccelerated;
+    return CGContextGetType(m_data->m_cgContext.get()) == kCGContextTypeIOSurface ? RenderingMode::Accelerated : RenderingMode::Unaccelerated;
```

Comment:
You said in the change log that "CoreAnimation might give a bitmap-backed CGContext for the drawing context, even though WebKit requested an accelerated context."

Did you check that `CGContextGetType()` will not return `kCGContextTypeIOSurface` for the context even if it was created by calling `CGIOSurfaceContextCreate`?

In other words, is it sufficient always to test the return value of `CGContextGetType()` to check whether the context is accelerated or not?

Reply:
>You said in the change log that "CoreAnimation might give a bitmap-backed CGContext for the drawing context, even though WebKit requested an accelerated context."

Yes, CoreAnimation does conceptually
```
if (size == small)
  return CGBitmapContextCreate();
return CGIOSurfaceContextCreate();
```

> Did you check that CGContextGetType() will not return kCGContextTypeIOSurface for the context even if it was created by calling CGIOSurfaceContextCreate?

CGContextGetType() is the "class type" accessor of the instance and it's used in many places to disambiguate the context type reliably. As far as I can read the code, yes, it's reliable.